### ISSUE TITLE
REST Block Renderer: Attributes registered using `blocks.registerBlockType` filter fails.

### ISSUE THREAD

**Issue by phpbits** [WordPress/gutenberg#15935]
Same issue with this one : https://github.com/WordPress/gutenberg/pull/13249 but not only fo className. All attributes registered using `blocks.registerBlockType` returns invalid on REST block renderer.

`{"code":"rest_invalid_param","message":"Invalid parameter(s): attributes","data":{"status":400,"params":{"attributes":"className is not a valid property of Object."}}}`

**Comment by phpbits** [comment]
https://github.com/WordPress/gutenberg/issues/15450

**Comment by talldan** [comment]
I don't think this is a bug, I expect the reason for this is that the attributes added via `registerBlockType` haven't been registered on the server, so `ServerSideRender` has no knowledge of them.

Registering the attributes server side as well (I think the `block_type_metadata` filter might be a candidate) should resolve the issue.

It's quite an old issue so I'll close it, if there's any further feedback though I can always reopen.
